server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/MailReprocessingIntegrationTest.java

```diff
@@ -188,6 +190,35 @@ public void reprocessingkShouldPreserveStateWhenProcessorIsNotSpecified() throws
         assertThat(containsExactlyOneMail(REPOSITORY_A)).isTrue();
     }
 
+    @Ignore("JAMES-2994 Reprocessing to an unknown processor triggers an infinite loop as the mail is nack")
+    @Test
+    public void reprocessingShouldProcessAsErrorWhenUnknownMailProcessor() throws Exception {
+        // Given an incoming email
+        messageSender.connect(LOCALHOST_IP, jamesServer.getProbe(SmtpGuiceProbe.class).getSmtpPort())
+            .sendMessage(FakeMail.builder()
+                .name("name")
+                .mimeMessage(MimeMessageBuilder.mimeMessageBuilder()
+                    .addToRecipient(RECIPIENT)
+                    .setSender(FROM)
+                    .setText("match me"))
+                .sender(FROM)
+                .recipient(RECIPIENT));
+        // Being stored in MailRepository B
+        awaitAtMostOneMinute.until(() -> containsExactlyOneMail(REPOSITORY_B));
+
+        // When I reprocess it
+        given()
+            .spec(specification)
+            .param("action", "reprocess")
+            .param("queue", MailQueueFactory.SPOOL)
+            .param("processor", "unknown")
+            .patch("/mailRepositories/" + REPOSITORY_B.getPath().urlEncoded() + "/mails").prettyPeek();
```

Comment:
The prettyPeek was for debug ? Or do you still want it ?

Reply:
No that's debug sorry